src/org/labkey/test/pages/core/login/DatabaseAuthConfigureDialog.java

```diff
@@ -0,0 +1,165 @@
+import org.openqa.selenium.WebElement;
+
+public class DatabaseAuthConfigureDialog extends AuthDialogBase<DatabaseAuthConfigureDialog>
+{
+
+    public DatabaseAuthConfigureDialog(LoginConfigRow row)
+    {
+        super(getFinder("Configure Database Authentication", row.getDriver()));
+    }
+
+    private static ModalDialogFinder getFinder(String title, WebDriver driver)
+    {
+        return new ModalDialogFinder(driver).withTitle(title);
+    }
+
+    // get password strength
+    public PasswordStrength getPasswordStrength()
+    {
+        if (elementCache().weakButton.getAttribute("class").contains("active"))
+            return PasswordStrength.Weak;
+        else return PasswordStrength.Strong;
+    }
+
+    public DatabaseAuthConfigureDialog setPasswordStrength(PasswordStrength desiredStrength)
+    {
+        if (getPasswordStrength().equals(desiredStrength))
+            return this;
+        else
+            if (desiredStrength.equals(PasswordStrength.Strong))
+                elementCache().strongButton.click();
+            else
+                elementCache().weakButton.click();
+        WebDriverWrapper.waitFor(()-> getPasswordStrength().equals(desiredStrength),
+                "the password strength was not set to desired state ["+desiredStrength.toString()+"] in time", 1000);
+        return this;
+    }
+
+    public DatabaseAuthConfigureDialog setPasswordExpiration(PasswordExpiration expiration)
+    {
+        elementCache().passwordExpirationSelect.selectOption(expiration);
+        return this;
+    }
+
+    public PasswordExpiration getPasswordExpiration()
+    {
+        return  PasswordExpiration.valueOf(elementCache().passwordExpirationSelect.getSelection().getValue());
+    }
+
+    @Override
+    public LoginConfigRow clickApply()
+    {
+        Locator.findAnyElement("Finish or Apply button", this,
+                Locators.dismissButton("Finish"), Locators.dismissButton("Apply")).click();
+        waitForClose(4);
+        return new LoginConfigRow.LoginConfigRowFinder(getDriver()).withDescription("Standard database authentication").waitFor();
+    }
+
+
+    @LogMethod
+    public void setDbLoginConfig(@LoggedParam PasswordStrength newStrength, PasswordExpiration newExpiration)
+    {
+        PasswordStrength curStrength = null;
+        PasswordExpiration curExpiration = null;
+
+        if (oldStrength == null || oldExpiration == null)
+        {
+            curStrength = getPasswordStrength();
+            curExpiration = getPasswordExpiration();
+        }
+
+        if ( newStrength != null && curStrength != newStrength)
+        {
+            if (oldStrength == null)
+                oldStrength = curStrength;
+            setPasswordStrength(newStrength);
+        }
+
+        if ( newExpiration != null && curExpiration != newExpiration)
+        {
+            if (oldExpiration == null)
+                oldExpiration = curExpiration;
+            setPasswordExpiration(newExpiration);
```

Comment:
Remember the old settings in the individual setters or make them private. If we want to have the ability to reset we need to make it so people can't easily bypass this logic.

Reply:
Okay.  I've split out the logic in that function and put the caching in the clickApply() override for this dialog